core/variant/array.cpp

```diff
@@ -202,53 +202,77 @@ uint32_t Array::recursive_hash(int recursion_count) const {
 }
 
 bool Array::_assign(const Array &p_array) {
-	bool can_convert = p_array._p->typed.type == Variant::NIL;
-	can_convert |= _p->typed.type == Variant::STRING && p_array._p->typed.type == Variant::STRING_NAME;
-	can_convert |= _p->typed.type == Variant::STRING_NAME && p_array._p->typed.type == Variant::STRING;
```

Comment:
I would be careful removing these lines! :) This was recently added by @rune-scape to ensure that `String` and `StringName` are unified. Maybe they'll have some more input on this.

Reply:
Here is [a comment with explanation](https://github.com/godotengine/godot/pull/69248#issuecomment-1345485373).

Reply:
I see - so because `can_convert_strict` works between `String` and `StringNames`, it isn't an issue? :)

Reply:
Correct.